### DISCUSSION TITLE
How to set password?

### DISCUSSION THREAD

**Discussion by yourplanner** [ChatGPTNextWeb/NextChat#1525] [Q&A] [answered]
Any Guide to ask user have to input default password then can access personal next web service?
請問有沒有教程可以知道怎麼設定密碼，讓使用者得先key密碼再繼續使用呢

**Comment by abicorios** [comment, marked as answer]
@yourplanner, do you mean OpenAI API key? It is this https://help.openai.com/en/articles/4936850-where-do-i-find-my-secret-api-key Also, in the `README.md` file of this project, in the `Get Started` section it is presented: https://github.com/Yidadaa/ChatGPT-Next-Web#get-started. 

And in settings you can past
![image](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/33332953/c3842fe9-c282-440b-b2e2-d962b05a5879)

If about Auth code, it is just any code:
![image](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/33332953/1235420e-a010-40fb-b488-49a17ee4c5a0)
![image](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/33332953/a8b26225-2dc7-467f-8455-d49ea97dd8f3)
![image](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/33332953/c3492d84-d7a0-45cd-b46f-8c6e5faf86d6)

Or do you mean some other? Like on adding the user login page? I can be like on these: https://blog.devso.io/implementing-credentials-provider-on-nextjs-and-nextauth, https://codevoweb.com/setup-and-use-nextauth-in-nextjs-13-app-directory, but it is not the simplest task, it can be a lot of little bugs to fix. It is often some old info, because `NextJS` is in updating from `pages` to `app` routing, and `NextAuth.js` is in updating to `Auth.js`.

**Comment by GetFractional** [comment]
I'm having an issue where that Auth page takes my code that I have set as an environment variable and opens up the Chatbot window again, but I get the following error message. Anyone know how to resolve this?

![image](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/134565741/135251a4-d062-4b52-ba4c-0f7470d1c264)

**Comment by Homqyy** [comment]
> Any Guide to ask user have to input default password then can access personal next web service?
> 
> 請問有沒有教程可以知道怎麼設定密碼，讓使用者得先key密碼再繼續使用呢

You can combine it with nginx that can provide a basic auth easily. The example of nginx.conf as following:

```
server {
    listen 443 ssl;

    server_name your.server.domain;

    ssl_certificate your.server.domain.crt;
    ssl_certificate_key your.server.domain.key;

    location / {
        auth_basic           "ChatGPT Next";
        auth_basic_user_file conf/htpasswd;
        proxy_pass http://backend.chatgpt;
    }    
}
```
Now, you can access to <https://your.server.domain>, but you will be demanded to provide user and password matched with credentials in `conf/htpasswd`.

More details in <https://nginx.org/en/docs/http/ngx_http_auth_basic_module.html>
